src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year4]) {
+    if (dateData[year2] < 30) {
+      dateData[year4] = `20${dateData[year2]}`;
+    } else {
+      dateData[year4] = `19${dateData[year2]}`;
+    }
+  } else {
+    dateData[year2] = dateData[year4].slice(2, 4);
```

Comment:
When slicing the year to get 'YY' from 'YYYY', it's safer to use '-2' as the starting index to ensure you're getting the last two digits regardless of the length of the year string.